src/scripts/main.js

```diff
@@ -1,3 +1,78 @@
 'use strict';
 
 // write code here
+const table = document.querySelector('table');
+const thead = document.querySelector('thead');
+const tbody = table.querySelector('tbody');
+const rows = [...table.querySelectorAll('tbody tr')];
+const headerCells = thead.querySelectorAll('th');
+
+headerCells[0].addEventListener('click', () => {
+  const sortedRows = rows.sort((rowA, rowB) => {
+    const nameA = rowA.querySelector('td:nth-child(1)').textContent.trim();
+    const nameB = rowB.querySelector('td:nth-child(1)').textContent.trim();
+
+    return nameA.localeCompare(nameB);
+  });
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => tbody.appendChild(row));
+});
+
+headerCells[1].addEventListener('click', () => {
+  const sortedRows = rows.sort((rowA, rowB) => {
+    const posA = rowA.querySelector('td:nth-child(2)').textContent.trim();
+    const posB = rowB.querySelector('td:nth-child(2)').textContent.trim();
+
+    return posA.localeCompare(posB);
+  });
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => tbody.appendChild(row));
+});
+
+headerCells[2].addEventListener('click', () => {
+  const sortedRows = rows.sort((rowA, rowB) => {
+    const ageA = parseInt(
+      rowA.querySelector('td:nth-child(3)').textContent.trim(),
+      10,
+    );
+    const ageB = parseInt(
+      rowB.querySelector('td:nth-child(3)').textContent.trim(),
+      10,
+    );
+
+    return ageA - ageB;
+  });
+
+  tbody.innerHTML = '';
+
+  sortedRows.forEach((row) => tbody.appendChild(row));
+});
+
+headerCells[3].addEventListener('click', () => {
+  const sortedRows = rows.sort((rowA, rowB) => {
```

Comment:
The `rows.sort()` method sorts the array in place. Consider creating a new array for sorting to preserve the original order for subsequent sorts.